cmd/tree-walk-pool.go

```diff
@@ -153,7 +164,10 @@ func (t *TreeWalkPool) Set(params listParams, resultCh chan TreeWalkResult, endW
 		t.pool[params] = append(walks, walkInfo)
 	} else {
 		// We are at limit, invalidate oldest, move list down and add new as last.
-		walks[0].endTimerCh <- struct{}{}
+		select {
+		case walks[0].endTimerCh <- struct{}{}:
```

Comment:
If I understood correctly, we can have some leaking go routines here, because
select {
case
default:
}

can ignore sending the message if the goroutine is not ready yet to listen from the channel

Reply:
@klauspost ^^

Reply:
It will only skip it if there is a message or it is closed. In both cases the goroutine will exit.

Reply:
@vadmeste @harshavardhana

Reply:
> It will only skip it if there is a message or it is closed. In both cases the goroutine will exit.

oh yes, because the channel is buffered